year2024/Day2/Day2Step1.java

```diff
@@ -0,0 +1,148 @@
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.stream.Stream;
+
+public class Day2Step1 {
+    private static final String FILE_NAME = ".\\src\\advent_of_code\\day2\\input.txt";
+    
+    public static void main(String[] args) throws IOException {
+        Day2Step1 day2 = new Day2Step1();
+        day2.day2();
+    }
+    
+    public void day2() throws IOException {
+        String[] allReports = getInput(FILE_NAME);
+        
+        // Part1, Part2
+        int safeCount = 0;
+        for (int i = 0; i < allReports.length; i++) {
+            int[] report = Stream.of(allReports[i].split(" "))
+                    .mapToInt(Integer::parseInt)
+                    .toArray();
+            /* Part1
+            if (!allLevelSafe(report)) {
+                continue;
+            }
+            */
+            // Part2
+            if (!allLevelSafe(report) && !isSafeWhenRemoveOneLevel(report)) {
+                continue;
+            }
+            safeCount++;
+        }
+        System.out.println(safeCount);
+    }
+
+    // initialize
+    private String[] getInput(String fileName) throws IOException {
+        Path inputFile = Paths.get(FILE_NAME).toAbsolutePath();
+        return Files.readString(inputFile).split("\r\n");

```

Comment:
`"\r\n"` は Windows の改行コードなので、他の OS では動かなさそうです。複数の改行コードに対応する正規表現を使って、OS ごとの差異がないようにするのがよさそうです

（動作確認してませんが）たぶん以下のようなコードにするとどの OS でも対応できそうです。

```suggestion
        return Files.readString(inputFile).split("\r\n|\n|\r");
```